vault_contract/src/contract.rs

```diff
@@ -47,6 +51,59 @@ impl VaultTrait for VaultContract {
 
         issuer::revoke_issuer(&e, &issuer, &did)
     }
+
+    fn store_vc(
+        e: Env,
+        vc_id: String,
+        vc_data: String,
+        recipient_did: String,
+        issuer_pk: Address,
+        issuance_contract_address: Address,
+    ) {
+        validate_did(&e, &recipient_did);
+        validate_issuer(&e, &issuer_pk, &recipient_did);
+
+        verifiable_credential::store_vc(
+            &e,
+            &vc_id,
+            &vc_data,
+            &issuance_contract_address,
+            &recipient_did,
+        );
+    }
+
+    fn get_vc(e: Env, vc_id: String) -> VerifiableCredential {
+        let vcs = storage::read_vcs(&e);
+
+        if !vcs.contains_key(vc_id.clone()) {
+            panic_with_error!(&e, ContractError::VCNotFound)
+        }
+        vcs.get_unchecked(vc_id.clone())
```

Comment:
@MarioRodriguezS @juanmagiraldor IMO we should use `Map::get` and a `match` statement to avoid calling `contains_key` and then `get_unchecked`